randomActivityGen.py

```diff
@@ -60,91 +59,108 @@
+            f"traffic: {outgoing_traffic}")
+        ET.SubElement(xml_gates, "entrance", attrib={
+            "edge": edge.getID(),
+            "incoming": str(incoming_traffic),
+            "outgoing": str(outgoing_traffic),
+            "pos": "0"
         })
 
 
+def verify_stats(stats: ET.ElementTree):
+    """
+    Do various verification on the stats file to ensure that it is usable. If population and work hours are missing,
+    some default values will be insert as these are required by ActivityGen.
+
+    :param stats: stats file parsed with ElementTree
+    """
+    city = stats.getroot()
+    assert city.tag == "city", "Stat file does not seem to be a valid stat file. The root element is not city"
+    # According to ActivityGen
+    # (https://github.com/eclipse/sumo/blob/master/src/activitygen/AGActivityGenHandler.cpp#L124-L161)
+    # only general::inhabitants and general::households are required. Everything else has default values.
+    general = stats.find("general")
+    # TODO Maybe guestimate the number of inhabitants and households based on the network's size
+    assert general is not None, "Stat file is missing <general>. Inhabitants and households are required"
+    assert general.attrib["inhabitants"] is not None, "Number of inhabitants are required"
+    assert general.attrib["households"] is not None, "Number of households are required"
+
+    # It is also required that there are at least one population bracket
+    population = city.find("population")
+    if population is None:
+        # Population is missing, so we add a default population
+        logging.info("Population is missing from statistics, adding a default configuration")
+        population = ET.SubElement(city, "population")
+        ET.SubElement(population, "bracket", {"beginAge": "0", "endAge": "30", "peopleNbr": "30"})
+        ET.SubElement(population, "bracket", {"beginAge": "30", "endAge": "60", "peopleNbr": "40"})
+        ET.SubElement(population, "bracket", {"beginAge": "60", "endAge": "90", "peopleNbr": "30"})
+
+    # Similarly at least and one opening and closing workhour is required
+    work_hours = city.find("workHours")
+    if work_hours is None:
+        # Work hours are missing, so we add some default work hours
+        logging.info("Work hours are missing from statistics, adding a default configuration")
+        work_hours = ET.SubElement(city, "workHours")
+        ET.SubElement(work_hours, "opening", {"hour": "28800", "proportion": "70"})  # 70% at 8.00
+        ET.SubElement(work_hours, "opening", {"hour": "30600", "proportion": "30"})  # 30% at 8.30
+        ET.SubElement(work_hours, "closing", {"hour": "43200", "proportion": "10"})  # 10% at 12.00
+        ET.SubElement(work_hours, "closing", {"hour": "61200", "proportion": "30"})  # 30% at 17.00
+        ET.SubElement(work_hours, "closing", {"hour": "63000", "proportion": "60"})  # 60% at 17.30
+
+
```

Comment:
These two functions are duplicate from `gates.py` and `utility`, respectively. Must've happened during merge from master.

Reply:
Yes, should be fixed now. e54ab46a9faab7f044f3db77ad2a00e099f2a25d and 3676551f21fc8693b52647596a11c23aa8e9d75d